CODING_STANDARDS.md

```diff
@@ -0,0 +1,158 @@
+# Coding Standards
+
+This is a working document outlining the coding standard for the general MariaDB codebase. It does not cover the coding standards for individual plugins, these should have their own coding standards documentation.
+
+## Git commit messages
+
+Git commit messages must conform to the 50/72 rule. This is is a long standard git standard which is automatically enforced by some editors. This means:
+
+* 50 characters max for the first (description) line.
+* A blank line
+* 72 characters max for every subsequent line.
+
+In addition if there is a Jira ticket number, this should be the first thing in the description. As an example:
+
+```
+MDEV-12345 Fixing Rockwell Turbo Encabulator
+
+The new principle involved is that instead of power being generated by
+the relative motion of conductors and fluxes, it’s produced by the
+modial interaction of magneto-reluctance and capacitive diractance.
+```
+
+The commit messages are typically rendered in Markdown format, so markdown formatting is permitted for the message body.
+
+## Coding Style (C / C++ files)
+
+Everyone has a preferred coding style, there is no real correct style for all projects around the world. What is important is that we stick to one common style throughout this code base.
+
+### Indentation
+
+We should use a variant of the [Allman indentation style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style). The variation is to use 2 spaces instead of tabs.
+
+Allman style specifies that braces associated with a statement should be on the following line with the same indentation and the statements inside the braces are next level indented. The closing braces are also on a new line at the same indentation as the original statement.
```

Comment:
What does the Allman style say about the `switch` statement? In our code base, the opening brace for `switch` is a special case:
```c
switch (x) {
case X:
```

Reply:
Hmm... yea, Allman isn't that. We also have a special case for 'if' as well by the look of Nikita's work I'll need to add.